app/main.py

```diff
@@ -1,2 +1,84 @@
+from copy import deepcopy
+
+from typing import Any
+
+
 class Dictionary:
-    pass
+
+    def __init__(self,) -> None:
+
+        self.length = 0
+        self.hash_table = [[] for _ in range(8)]
+
+    def __setitem__(self, key: Any, value: Any) -> None:
+        threshold = (len(self.hash_table) * 2) // 3
+        if self.length == threshold:
+            self.resize()
+        hash_key = hash(key)
+        index = self.get_index(hash_key)
+
+        if self.hash_table[index]:
+
+            while self.hash_table[index]:
+                if key == self.hash_table[index][0]:
+                    self.hash_table[index][1] = value
+                    return
+                index += 1
+                if index == len(self.hash_table):
+                    index = 0
+            self.hash_table[index] = [key, value, key.__hash__()]
```

Comment:
You can use  variable hash_key, no need to use hash function again

Reply:
It does not work without key.__hash__() or hash(key)